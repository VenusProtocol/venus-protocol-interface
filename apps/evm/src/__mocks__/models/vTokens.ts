import type { VToken } from 'types';

import { bnb, busd, eth, lisUsd, luna, usdc, usdt, ust, wbnb, weth, xvs } from './tokens';

export const vXvs: VToken = {
  address: '0x6d6F697e34145Bb95c54E77482d97cc261Dc237E',
  decimals: 8,
  symbol: 'vXVS',
  underlyingToken: xvs,
};

export const vBnb: VToken = {
  address: '0x2E7222e51c0f6e98610A1543Aa3836E092CDe62c',
  decimals: 8,
  symbol: 'vBNB',
  underlyingToken: bnb,
};

export const vUsdc: VToken = {
  address: '0xD5C4C2e2facBEB59D0216D0595d63FcDc6F9A1a7',
  decimals: 8,
  symbol: 'vUSDC',
  underlyingToken: usdc,
};

export const vUsdtCorePool: VToken = {
  address: '0xb7526572FFE56AB9D7489838Bf2E18e3323b441A',
  decimals: 8,
  symbol: 'vUSDT',
  underlyingToken: usdt,
};

export const vUsdtDeFi: VToken = {
  address: '0x80CC30811e362aC9aB857C3d7875CbcCc0b65750',
  decimals: 8,
  symbol: 'vUSDT_DeFi',
  underlyingToken: usdt,
};

export const vUsdtGameFi: VToken = {
  address: '0x0bFE4e0B8A2a096A27e5B18b078d25be57C08634',
  decimals: 8,
  symbol: 'vUSDT_GameFi',
  underlyingToken: usdt,
};

export const vUsdtLiquidStakedBnb: VToken = {
  address: '0x2197d02cC9cd1ad51317A0a85A656a0c82383A7c',
  decimals: 8,
  symbol: 'vUSDT_Liquid_Staked_BNB',
  underlyingToken: usdt,
};

export const vUsdtMeme: VToken = {
  address: '0x3AF2bE7AbEF0f840b196D99d79F4B803a5dB14a1',
  decimals: 8,
  symbol: 'vUSDT_Meme',
  underlyingToken: usdt,
};

export const vUsdtStablecoins: VToken = {
  address: '0x3338988d0beb4419Acb8fE624218754053362D06',
  decimals: 8,
  symbol: 'vUSDT_Stablecoins',
  underlyingToken: usdt,
};

export const vUsdtTron: VToken = {
  address: '0x712774CBFFCBD60e9825871CcEFF2F917442b2c3',
  decimals: 8,
  symbol: 'vUSDT_Tron',
  underlyingToken: usdt,
};

export const vBusdCorePool: VToken = {
  address: '0x08e0A5575De71037aE36AbfAfb516595fE68e5e4',
  decimals: 8,
  symbol: 'vBUSD',
  underlyingToken: busd,
};

export const vBusdStablecoins: VToken = {
  address: '0x8c8A1a0b6e1cb8058037F7bF24de6b79Aca5B7B0',
  decimals: 8,
  symbol: 'vBUSD',
  underlyingToken: busd,
};

export const vUst: VToken = {
  address: '0xF206af85BC2761c4F876d27Bd474681CfB335EfA',
  decimals: 8,
  symbol: 'vUST',
  underlyingToken: ust,
};

export const vLuna: VToken = {
  address: '0x9C3015191d39cF1930F92EB7e7BCbd020bCA286a',
  decimals: 8,
  symbol: 'vLUNA',
  underlyingToken: luna,
};

export const vWeth: VToken = {
  address: '0xc2931B1fEa69b6D6dA65a50363A8D75d285e4da9',
  decimals: 8,
  symbol: 'vWETH',
  underlyingToken: weth,
};

export const vEth: VToken = {
  address: '0x86f8DfB7CA84455174EE9C3edd94867b51Da46BD',
  decimals: 8,
  symbol: 'vETH',
  underlyingToken: eth,
};

export const vLisUSD: VToken = {
  address: '0x170d3b2da05cc2124334240fB34ad1359e34C562',
  decimals: 8,
  symbol: 'vlisUSD',
  underlyingToken: lisUsd,
};

export const vWbnb: VToken = {
  address: '0x231dED0Dfc99634e52EE1a1329586bc970d773b3',
  decimals: 8,
  symbol: 'vWBNB',
  underlyingToken: wbnb,
};

export default [
  vXvs,
  vBnb,
  vUsdc,
  vUsdtCorePool,
  vUsdtDeFi,
  vUsdtGameFi,
  vUsdtLiquidStakedBnb,
  vUsdtMeme,
  vUsdtStablecoins,
  vUsdtTron,
  vBusdCorePool,
  vBusdStablecoins,
  vUst,
  vLuna,
  vEth,
  vLisUSD,
  vWbnb,
];
