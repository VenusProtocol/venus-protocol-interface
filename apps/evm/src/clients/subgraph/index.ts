export * from './queries/isolatedPools/getIsolatedPoolParticipantsCount';
export * from './queries/governance/getProposals';
