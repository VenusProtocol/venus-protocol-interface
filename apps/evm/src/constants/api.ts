import type { Network } from 'types';

export const apiUrls: Record<Network, string> = {
  testnet: 'https://testnetapi.venus.io',
  mainnet: 'https://api.venus.io',
  'mainnet-preview': 'https://api-preview.venus.io',
};
