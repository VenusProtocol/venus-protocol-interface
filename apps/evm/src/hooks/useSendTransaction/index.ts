import { ContractReceipt, ContractTransaction } from 'ethers';
import { MutationObserverOptions, useMutation } from 'react-query';

import FunctionKey from 'constants/functionKey';
import { TransactionType } from 'types';

import { CONFIRMATIONS, useTrackTransaction } from './useTrackTransaction';

export interface UseSendTransactionOptions<TMutateInput extends Record<string, unknown> | void>
  extends MutationObserverOptions<unknown, Error, TMutateInput> {
  waitForConfirmation?: boolean;
}

export interface UseSendTransactionInput<TMutateInput extends Record<string, unknown> | void> {
  fn: (input: TMutateInput) => Promise<ContractTransaction>;
  fnKey: FunctionKey | [FunctionKey, ...unknown[]];
  transactionType?: TransactionType;
  onConfirmed?: (input: {
    transaction: ContractTransaction;
    transactionReceipt: ContractReceipt;
    input: TMutateInput;
  }) => Promise<unknown> | unknown;
  onReverted?: (input: {
    transaction: ContractTransaction;
    input: TMutateInput;
  }) => Promise<unknown> | unknown;
  options?: UseSendTransactionOptions<TMutateInput>;
}

export const useSendTransaction = <TMutateInput extends Record<string, unknown> | void>({
  fn,
  fnKey,
  transactionType,
  onConfirmed,
  onReverted,
  options,
}: UseSendTransactionInput<TMutateInput>) => {
  const trackTransaction = useTrackTransaction({ transactionType });

  return useMutation(
    fnKey,
    async input => {
      // Send transaction
      const transaction = await fn(input);

      // Track transaction's progress in the background
      trackTransaction({
        transaction,
        onConfirmed: onConfirmedInput => onConfirmed?.({ ...onConfirmedInput, input }),
        onReverted: onRevertedInput => onReverted?.({ ...onRevertedInput, input }),
      });

      if (options?.waitForConfirmation) {
        // Only return when transaction has been confirmed
        await transaction.wait(CONFIRMATIONS);
      }
    },
    options,
  );
};
