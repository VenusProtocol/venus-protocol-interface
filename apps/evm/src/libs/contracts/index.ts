export * from './generated/addresses';
export * from './generated/abis';

export * from './utilities/getContractAddress';
