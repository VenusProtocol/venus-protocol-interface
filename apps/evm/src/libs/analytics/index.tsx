import { PostHogProvider, usePostHog } from 'posthog-js/react';
import { useEffect } from 'react';

import config from 'config';
import { version as APP_VERSION } from 'constants/version';
import { useAccountAddress, useChainId } from 'libs/wallet';

const UserIdentifier: React.FC = () => {
  const { accountAddress } = useAccountAddress();
  const { chainId } = useChainId();
  const posthog = usePostHog();

  // Identify user by their account address along with the network chain they are on
  useEffect(() => {
    if (accountAddress) {
      posthog?.identify(accountAddress, { chainId });
    } else {
      posthog?.reset();
    }
  }, [accountAddress, chainId, posthog]);

  return null;
};

export interface AnalyticProviderProps {
  children?: React.ReactNode;
}

export const AnalyticProvider: React.FC<AnalyticProviderProps> = ({ children }) => (
  <PostHogProvider
    apiKey={config.posthog.apiKey}
    options={{
      api_host: config.posthog.hostUrl,
      persistence: 'memory',
      name: APP_VERSION,
    }}
  >
    <UserIdentifier />

    {children}
  </PostHogProvider>
);

export * from './useAnalytics';
export { default as useAnalytics } from './useAnalytics';
