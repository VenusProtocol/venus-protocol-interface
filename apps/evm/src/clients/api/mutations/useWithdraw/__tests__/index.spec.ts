import fakeAccountAddress, {
  altAddress as fakePoolComptrollerContractAddress,
} from '__mocks__/models/address';
import { vBnb, vUsdt, vWeth } from '__mocks__/models/vTokens';
import BigNumber from 'bignumber.js';
import { queryClient } from 'clients/api';
import { useSendTransaction } from 'hooks/useSendTransaction';
import { useAnalytics } from 'libs/analytics';
import { getContractAddress } from 'libs/contracts';
import { usePublicClient } from 'libs/wallet';
import { renderHook } from 'testUtils/render';
import type { Mock } from 'vitest';
import { useWithdraw } from '..';

vi.mock('libs/contracts');

const fakeAmountMantissa = new BigNumber('10000000000000000');

const fakeInput = {
  vToken: vUsdt,
  amountMantissa: fakeAmountMantissa,
  poolComptrollerContractAddress: fakePoolComptrollerContractAddress,
  poolName: 'Fake Pool',
};

const fakeOptions = {
  gasless: true,
  waitForConfirmation: true,
};

const mockCaptureAnalyticEvent = vi.fn();

describe('useWithdraw', () => {
  beforeEach(() => {
    (usePublicClient as Mock).mockImplementation(() => ({
      publicClient: {
        createAccessList: vi.fn(() => ({
          accessList: 'fake-access-list',
        })),
      },
    }));

    (useAnalytics as Mock).mockImplementation(() => ({
      captureAnalyticEvent: mockCaptureAnalyticEvent,
    }));
  });

  it('calls useSendTransaction with the correct parameters when withdrawing tokens', async () => {
    renderHook(() => useWithdraw(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    expect(useSendTransaction).toHaveBeenCalledWith({
      fn: expect.any(Function),
      onConfirmed: expect.any(Function),
      options: fakeOptions,
    });

    const { fn, onConfirmed } = (useSendTransaction as Mock).mock.calls[0][0];

    expect(await fn(fakeInput)).toMatchInlineSnapshot(
      {
        abi: expect.any(Array),
      },
      `
      {
        "abi": Any<Array>,
        "accessList": undefined,
        "address": "0xb7526572FFE56AB9D7489838Bf2E18e3323b441A",
        "args": [
          10000000000000000n,
        ],
        "functionName": "redeemUnderlying",
      }
    `,
    );

    onConfirmed({ input: fakeInput });

    expect(mockCaptureAnalyticEvent.mock.calls[0]).toMatchInlineSnapshot(`
      [
        "Tokens withdrawn",
        {
          "poolName": "Fake Pool",
          "tokenAmountTokens": 10000000000,
          "tokenSymbol": "USDT",
          "withdrewFullSupply": false,
        },
      ]
    `);

    expect((queryClient.invalidateQueries as Mock).mock.calls).toMatchSnapshot();
  });

  it('calls useSendTransaction with the correct parameters when withdrawing full supply', async () => {
    renderHook(() => useWithdraw(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    const withdrawFullSupplyInput = {
      ...fakeInput,
      withdrawFullSupply: true,
    };

    expect(useSendTransaction).toHaveBeenCalledWith({
      fn: expect.any(Function),
      onConfirmed: expect.any(Function),
      options: fakeOptions,
    });

    const { fn, onConfirmed } = (useSendTransaction as Mock).mock.calls[0][0];

    expect(await fn(withdrawFullSupplyInput)).toMatchInlineSnapshot(
      {
        abi: expect.any(Array),
      },
      `
      {
        "abi": Any<Array>,
        "accessList": undefined,
        "address": "0xb7526572FFE56AB9D7489838Bf2E18e3323b441A",
        "args": [
          10000000000000000n,
        ],
        "functionName": "redeem",
      }
    `,
    );

    onConfirmed({ input: withdrawFullSupplyInput });

    expect(mockCaptureAnalyticEvent.mock.calls[0]).toMatchInlineSnapshot(`
      [
        "Tokens withdrawn",
        {
          "poolName": "Fake Pool",
          "tokenAmountTokens": 10000000000,
          "tokenSymbol": "USDT",
          "withdrewFullSupply": false,
        },
      ]
    `);

    expect((queryClient.invalidateQueries as Mock).mock.calls).toMatchSnapshot();
  });

  it('calls useSendTransaction with the correct parameters when withdrawing and unwrapping tokens', async () => {
    renderHook(() => useWithdraw(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    const withdrawAndUnwrapInput = {
      ...fakeInput,
      vToken: vWeth,
      unwrap: true,
    };

    expect(useSendTransaction).toHaveBeenCalledWith({
      fn: expect.any(Function),
      onConfirmed: expect.any(Function),
      options: fakeOptions,
    });

    const { fn, onConfirmed } = (useSendTransaction as Mock).mock.calls[0][0];

    expect(await fn(withdrawAndUnwrapInput)).toMatchInlineSnapshot(
      {
        abi: expect.any(Array),
      },
      `
      {
        "abi": Any<Array>,
        "address": "0xfakeNativeTokenGatewayContractAddress",
        "args": [
          10000000000000000n,
        ],
        "functionName": "redeemUnderlyingAndUnwrap",
      }
    `,
    );

    onConfirmed({ input: withdrawAndUnwrapInput });

    expect(mockCaptureAnalyticEvent.mock.calls[0]).toMatchInlineSnapshot(`
      [
        "Tokens withdrawn",
        {
          "poolName": "Fake Pool",
          "tokenAmountTokens": 0.01,
          "tokenSymbol": "WETH",
          "withdrewFullSupply": false,
        },
      ]
    `);

    expect((queryClient.invalidateQueries as Mock).mock.calls).toMatchSnapshot();
  });

  it('calls useSendTransaction with the correct parameters when withdrawing full supply and unwrapping', async () => {
    renderHook(() => useWithdraw(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    const withdrawFullSupplyAndUnwrapInput = {
      ...fakeInput,
      vToken: vBnb,
      withdrawFullSupply: true,
      unwrap: true,
    };

    expect(useSendTransaction).toHaveBeenCalledWith({
      fn: expect.any(Function),
      onConfirmed: expect.any(Function),
      options: fakeOptions,
    });

    const { fn, onConfirmed } = (useSendTransaction as Mock).mock.calls[0][0];

    expect(await fn(withdrawFullSupplyAndUnwrapInput)).toMatchInlineSnapshot(
      {
        abi: expect.any(Array),
      },
      `
      {
        "abi": Any<Array>,
        "address": "0xfakeNativeTokenGatewayContractAddress",
        "args": [
          10000000000000000n,
        ],
        "functionName": "redeemAndUnwrap",
      }
    `,
    );

    onConfirmed({ input: withdrawFullSupplyAndUnwrapInput });

    expect(mockCaptureAnalyticEvent.mock.calls[0]).toMatchInlineSnapshot(`
      [
        "Tokens withdrawn",
        {
          "poolName": "Fake Pool",
          "tokenAmountTokens": 0.01,
          "tokenSymbol": "BNB",
          "withdrewFullSupply": false,
        },
      ]
    `);

    expect((queryClient.invalidateQueries as Mock).mock.calls).toMatchSnapshot();
  });

  it('calls useSendTransaction with the correct parameters when withdrawing native token', async () => {
    renderHook(() => useWithdraw(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    const withdrawNativeInput = {
      ...fakeInput,
      vToken: vBnb,
    };

    expect(useSendTransaction).toHaveBeenCalledWith({
      fn: expect.any(Function),
      onConfirmed: expect.any(Function),
      options: fakeOptions,
    });

    const { fn, onConfirmed } = (useSendTransaction as Mock).mock.calls[0][0];

    expect(await fn(withdrawNativeInput)).toMatchInlineSnapshot(
      {
        abi: expect.any(Array),
      },
      `
      {
        "abi": Any<Array>,
        "accessList": "fake-access-list",
        "address": "0x2E7222e51c0f6e98610A1543Aa3836E092CDe62c",
        "args": [
          10000000000000000n,
        ],
        "functionName": "redeemUnderlying",
      }
    `,
    );

    onConfirmed({ input: withdrawNativeInput });

    expect(mockCaptureAnalyticEvent.mock.calls[0]).toMatchInlineSnapshot(`
      [
        "Tokens withdrawn",
        {
          "poolName": "Fake Pool",
          "tokenAmountTokens": 0.01,
          "tokenSymbol": "BNB",
          "withdrewFullSupply": false,
        },
      ]
    `);

    expect((queryClient.invalidateQueries as Mock).mock.calls).toMatchSnapshot();
  });

  it('throws when account address is not available', async () => {
    renderHook(() => useWithdraw(fakeOptions));

    const { fn } = (useSendTransaction as Mock).mock.calls[0][0];

    expect(async () => fn(fakeInput)).rejects.toThrow('somethingWentWrong');
  });

  it('throws when native token gateway contract address is not available for unwrap', async () => {
    (getContractAddress as Mock).mockImplementation(() => undefined);

    renderHook(() => useWithdraw(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    const { fn } = (useSendTransaction as Mock).mock.calls[0][0];

    const withdrawAndUnwrapInput = {
      ...fakeInput,
      vToken: vWeth,
      unwrap: true,
    };

    expect(async () => fn(withdrawAndUnwrapInput)).rejects.toThrow('somethingWentWrong');
  });
});
