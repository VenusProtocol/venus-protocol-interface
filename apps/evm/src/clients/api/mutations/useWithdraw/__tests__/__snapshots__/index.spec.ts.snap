// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useWithdraw > calls useSendTransaction with the correct parameters when withdrawing and unwrapping tokens 3`] = `
[
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCES_ALL",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_POOLS",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCE",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "vTokenAddress": "0xc2931B1fEa69b6D6dA65a50363A8D75d285e4da9",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_TOKEN_BALANCES",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_BALANCE_OF",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "tokenAddress": "0x700868CAbb60e90d77B6588ce072d9859ec8E281",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_BALANCE_OF",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "tokenAddress": "0x98f7A83361F7Ac8765CcEBAB1425da6b341958a7",
        },
      ],
    },
  ],
]
`;

exports[`useWithdraw > calls useSendTransaction with the correct parameters when withdrawing full supply 3`] = `
[
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCES_ALL",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_POOLS",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCE",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "vTokenAddress": "0xb7526572FFE56AB9D7489838Bf2E18e3323b441A",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_TOKEN_BALANCES",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_BALANCE_OF",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "tokenAddress": "0xA11c8D9DC9b66E209Ef60F0C8D969D3CD988782c",
        },
      ],
    },
  ],
]
`;

exports[`useWithdraw > calls useSendTransaction with the correct parameters when withdrawing full supply and unwrapping 3`] = `
[
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCES_ALL",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_POOLS",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCE",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "vTokenAddress": "0x2E7222e51c0f6e98610A1543Aa3836E092CDe62c",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_TOKEN_BALANCES",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_BALANCE_OF",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "tokenAddress": "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
        },
      ],
    },
  ],
]
`;

exports[`useWithdraw > calls useSendTransaction with the correct parameters when withdrawing native token 3`] = `
[
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCES_ALL",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_POOLS",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCE",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "vTokenAddress": "0x2E7222e51c0f6e98610A1543Aa3836E092CDe62c",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_TOKEN_BALANCES",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_BALANCE_OF",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "tokenAddress": "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
        },
      ],
    },
  ],
]
`;

exports[`useWithdraw > calls useSendTransaction with the correct parameters when withdrawing tokens 3`] = `
[
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCES_ALL",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_POOLS",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCE",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "vTokenAddress": "0xb7526572FFE56AB9D7489838Bf2E18e3323b441A",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_TOKEN_BALANCES",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_BALANCE_OF",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "tokenAddress": "0xA11c8D9DC9b66E209Ef60F0C8D969D3CD988782c",
        },
      ],
    },
  ],
]
`;
